ClosedXML/Excel/AutoFilters/XLAutoFilter.cs

```diff
@@ -144,14 +146,18 @@ public XLAutoFilter Clear()
 
         public XLAutoFilter Set(IXLRangeBase range)
         {
+            var firstOverlappingTable = range.Worksheet.Tables.FirstOrDefault(t => t.RangeUsed().Intersects(range));
```

Comment:
I guess we can safely assume there will not be many hundreds of table on a single worksheet, so this way of searching is efficient enough.

Reply:
On .NET Framework, the evaluation will not continue past the first match, so it's similar to the old approach. Apparently this changed on .NET Core.

Reply:
I don't think this changed in .Net Core; either way, it won't stop until reaching the end if there are no intersections. And if we'd have thousands of tables on the same worksheet the search would be relatively slow.
But since there are usually just a few tables on a worksheet I think this is not a problem.

Reply:
I thought of this discussion that I saw the other day. https://github.com/dotnet/runtime/issues/31554
But seems it's applicable only on ordered enumerables. Still something to take note of.